compiler-rt/lib/sanitizer_common/sanitizer_linux.cpp

```diff
@@ -2118,8 +2118,91 @@ bool SignalContext::IsTrueFaultingAddress() const {
   return si->si_signo == SIGSEGV && si->si_code != 128;
 }
 
+UNUSED
+static const char *RegNumToRegName(int reg) {
+#  if defined(__x86_64__)
+  switch (reg) {
+    case REG_RAX:
+      return "rax";
+    case REG_RBX:
+      return "rbx";
+    case REG_RCX:
+      return "rcx";
+    case REG_RDX:
+      return "rdx";
+    case REG_RDI:
+      return "rdi";
+    case REG_RSI:
+      return "rsi";
+    case REG_RBP:
+      return "rbp";
+    case REG_RSP:
+      return "rsp";
+    case REG_R8:
+      return "r8";
+    case REG_R9:
+      return "r9";
+    case REG_R10:
+      return "r10";
+    case REG_R11:
+      return "r11";
+    case REG_R12:
+      return "r12";
+    case REG_R13:
+      return "r13";
+    case REG_R14:
+      return "r14";
+    case REG_R15:
+      return "r15";
+    default:
+      return NULL;
+  }
+#  endif
+  return NULL;
+}
+
 void SignalContext::DumpAllRegisters(void *context) {
-  // FIXME: Implement this.
+#  if SANITIZER_LINUX
+  ucontext_t *ucontext = (ucontext_t *)context;
+#    define DUMPREG64(r)                             \
+      Printf("%s = 0x%016llx  ", RegNumToRegName(r), \
```

Comment:
mac version uses preprocessor for `#r`
if you replace that with **RegNumToRegName**, we don't need macro any more
lambda or function will be nicer.

Reply:
Also can we just move  SignalContext::DumpAllRegisters from MAC ?

Reply:
Yeah, i thought about it.
copy-paste is not possible because of some difference in ucontext implementation for linux and mac.
And what should we write after `REQUIRES:` in the test if there will be an implementation for arm-linux, i386-linux, x86_64-linux and aarch64-linux? Can we require only `linux`?

Reply:
The differences in ucontext definition is the reason why i made RegNumToRegName

Reply:
I think that i will handle your suggestion about removing macroses from the `DumpAllRegisters` code for x86-64 and i386 but arm and aarch64 implementations will be done later because there is a need to change MAC's code and i don't have arm/aarch64 environment to test it :(